zzk/service/hoststate.go

```diff
@@ -50,19 +51,32 @@ type HostStateHandler interface {
 	PauseContainer(serviceID string, instanceID int) error
 }
 
+type HostStateThread struct {
```

Comment:
This should be private, you may even want to just embed this into HostStateListener, i.e.
```
threads map[string]struct{
data *ServiceState
exited <-chan time.Time
}
```